drivers/sensor/fxos8700/fxos8700.c

```diff
@@ -332,20 +332,21 @@ static int fxos8700_init(struct device *dev)
 		return -EIO;
 	}
 
+	k_sem_init(&data->sem, 1, UINT_MAX);
```

Comment:
You can instead initialize the count to zero and remove the `k_sem_take` in line 343.